src/datapilot/utils/utils.py

```diff
@@ -58,3 +68,256 @@ def is_superset_path(superset_path: str, path: str):
+                    )
+                )
+    return nodes
+
+
+def get_manifest_source_nodes(manifest: Dict, sources: List) -> List[SourceNode]:
+    nodes = []
+    for node in manifest["sources"].values():
+        if node["source_name"] in sources:
+            nodes.append(
+                SourceNode(
+                    unique_id=node["unique_id"],
+                    name=node["source_name"],
+                    resource_type=node["resource_type"],
+                    table=node["identifier"],
+                    database=node["database"],
+                    table_schema=node["schema"],
+                )
+            )
+    return nodes
+
+
+def get_model_tables(models: List[ModelNode]) -> List[str]:
+    tables = []
+    for model in models:
+        tables.append(f"{model.database}.{model.table_schema}.{model.alias}")
+    return tables
+
+
+def get_source_tables(sources: List[SourceNode]) -> List[str]:
+    tables = []
+    for source in sources:
+        tables.append(f"{source.database}.{source.table_schema}.{source.name}")
+    return tables
+
+
+def get_table_name(node: Union[ModelNode, SourceNode], node_type: str) -> str:
+    if node_type == "nodes":
+        return f"{node.database}.{node.table_schema}.{node.alias}"
+    return f"{node.database}.{node.table_schema}.{node.name}"
+
+
+def fill_catalog(table_columns_map: Dict, manifest: Dict, catalog: Dict, nodes: List[Union[ModelNode, SourceNode]], node_type: str) -> Dict:
+    if not nodes:
+        catalog[node_type] = {}
+        return catalog
+
+    for node in nodes:
+        columns = {}
+        for column in table_columns_map[node.unique_id]:
+            column_type = get_column_type(column["dtype"])
+            columns[column["column"]] = {
+                "type": column_type,
+                "index": len(columns) + 1,
+                "name": column["column"],
+                "comment": None,
+            }
+
+        catalog[node_type] = {
+            node.unique_id: {
+                "metadata": {
+                    "type": "BASE TABLE",
+                    "schema": manifest[node_type][node.unique_id]["schema"],
+                    "name": node.alias if node_type == "nodes" else node.name,
+                    "database": manifest[node_type][node.unique_id]["database"],
+                    "comment": None,
+                    "owner": None,
+                },
+                "columns": columns,
+                "stats": {},
+                "unique_id": node.unique_id,
+            }
+        }
+
+    return catalog
+
+
+def run_macro(macro: str, base_path: str) -> str:
+    dbt_compile = subprocess.run(
+        ["dbt", "compile", "--inline", macro],  # noqa
+        capture_output=True,
+        cwd=base_path,
+        text=True,
+    )
+    return dbt_compile.stdout
+
+
+def generate_partial_manifest_catalog(changed_files, manifest_path: str, catalog_path: str, base_path: str = "./"):
+    try:
```

Comment:
@mdesmet Here is where the core logic is

Reply:
@mdesmet If you know better ways to print results, feel free to share

Reply:
I think it is fine, printing here is not really logging it is the intended output.